src/Accounts/Accounts/Models/RMProfileClient.cs

```diff
@@ -325,12 +322,12 @@ public bool TryRemoveContext(IAzureContext context)
                         }
 
                         if (token != null &&
-                            defaultTenant == null &&
+                            (defaultTenant == null || selectSubscriptionFromList) &&
```

Comment:
This block is used to obtain default subscription. 
- If selectSubscriptionFromList is false, return the first available subscription in the first available tenant, will not continue listing subscriptions from another tenants.
- If selectSubscriptionFromList is true, needs to list all available subscriptions to allow further selection.